### PR TITLE
[enh] add halt feature:

### PR DESCRIPTION
**Pull request by M5oul**
- Retrieve running VM.
- Turn the VM off if running.
- Otherwise warn that there is no running VM.

### CODE REVIEW

File: ynh-dev [hunk 1]
```diff
@@ -145,6 +147,21 @@ elif [ "$1" = "run" ]; then
 
 
 #####################
+## Halt running VM ##
+#####################
+elif [ "$1" = "halt" ]; then
+    vm=$(vagrant global-status | head -3 | tail -1)
```
Comment (opi) [inline ynh-dev]: This returns only the last run Vagrant Box, how can we be sure that is the good one ?

File: ynh-dev [hunk 2]
```diff
@@ -145,6 +147,21 @@ elif [ "$1" = "run" ]; then
...
+    vm_status=$(echo $vm | cut -d ' ' -f 4)
```
Comment (opi) [inline ynh-dev]: on my host (debian jessie, Vagrant 1.6.5), `cut -d ' ' -f 4` does not return anything.

Here is what $vm contains : `07c029b  old-stable      virtualbox poweroff /path/to/yunohost/devenv/stable`; MAybe playing with this good old `awk` command would be more robust.

File: ynh-dev (hunk 1 above)
Comment (M5oul) [inline ynh-dev]: Good question.
My code is a bit dirty as it consider there is only one VM.

Comment (opi) [inline ynh-dev]: Maye by we can mimic what the `ynh-dev run` command do, by adding an parameter with version name. That way we can grep the output of `vagrant global-status`

**Comment by M5oul** [comment]
Wow, I forgot to document it on the readme.
Will do it asap.

**Comment by Psycojoker** [comment]
up

**Comment by M5oul** [comment]
The readme changed too much.
I documented it with this PR https://github.com/YunoHost/ynh-dev/pull/16.

**Comment by M5oul** [comment]
Are you agree that this command only halt one VM?
We usually use one VM. That's have always been my case.
If someone want to had multi-VM halt support. I think it must be done later.

I don't know if it's similar subcmd has [`suspend` cmd](https://github.com/YunoHost/ynh-dev#16-suspend-or-kill-your-vm).
I didn't tested `suspend` subcmd, or it wasn't working. I can't remember.

**Comment by Psycojoker** [comment]
I generally uses `vagrant halt` that directly does this job so I don't see the point of this new command but I won't oppose it.

**Comment by zamentur** [comment]
Like Bram

File: ynh-dev (hunk 1 above)
Comment (zamentur) [inline ynh-dev]: This code don't work. On my host I have several vagrant (for works and not only yunohost) and in my case it's an other vm which is returned by this line. This implementation could be dangerous.

**Comment by M5oul** [comment]
I close it as in fact, we could have many boxes and this code won't works.

**Pull request closed without merging**